src/main/java/org/primefaces/PrimeFaces.java

```diff
@@ -359,4 +383,112 @@ public void update(String... expressions) {
             update(Arrays.asList(expressions));
         }
     }
+
+    public class MultiViewState {
+
+        private static final String SEPARATOR = "_";
+
+        /**
+         * Removes all multiViewState within the current session.
+         */
+        public void clearMultiViewStates() {
+            getFacesContext().getExternalContext().getSessionMap().remove(Constants.MULTI_VIEW_STATES);
+        }
+
+        /**
+         * Removes all multiViewState in specific view within the current session.
+         *
+         * @param viewId viewId in which multiview state should be cleared
+         */
+        public void clearMultiViewStates(String viewId) {
+            clearMultiViewStates(viewId, null);
+        }
+
+        /**
+         * Removes all multiViewState in specific view within the current session.
+         *
+         * @param viewId viewId in which multiview state should be cleared
+         * @param clientIdConsumer operation to execute for every clientId after multiview state has been cleared
+         */
+        public void clearMultiViewStates(String viewId, Consumer<String> clientIdConsumer) {
+            String stateKey = createViewId(viewId);
+            Map<String, Object> multiViewStates = getMultiViewStateSessionMap();
+            Set<String> states = multiViewStates.keySet().stream()
+                    .filter(s -> s.startsWith(stateKey))
+                    .collect(Collectors.toSet());
+            if (!states.isEmpty()) {
+                multiViewStates.keySet().removeAll(states);
+
+                if (clientIdConsumer != null) {
+                    states.stream()
+                            .map(s -> s.replace(stateKey + SEPARATOR, ""))
+                            .forEach(clientIdConsumer);
```

Comment:
Use Constants.EMPTY_STRING instead of "".

Reply:
Good point